protobuf/src/main/java/io/grpc/protobuf/ProtoUtils.java

```diff
@@ -107,7 +107,8 @@ private T parseFrom(InputStream stream) throws InvalidProtocolBufferException {
    */
   public static <T extends Message> Metadata.Key<T> keyForProto(final T instance) {
     return Metadata.Key.of(
-        instance.getDescriptorForType().getFullName() + Metadata.BINARY_HEADER_SUFFIX,
+        instance.getDescriptorForType().getFullName().replace('.', '-')
```

Comment:
@carl-mastrangelo, let's go into a holding pattern and figure out what end result we are most favorable toward. I think Louis and I could quickly get to a point which we would know where we stand enough to decide what to do with this changed line.

@louiscryan, what are your thoughts on the metadata key names to use for protobuf? It seems before we were using period in the name, which now is disallowed.